docs/how-to/setting-up-type-safety.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Setting up type safety
+---
+
+To know more about how type safety works in React Router, check out our [dedicated explanation](../explanation//type-safety.md).
+
+# Setting up type safety
+
+React Router generates types for into a `.react-router/` directory at the root of your app.
+This directory is fully managed by React Router and is derived from your route config (`app/routes.ts` by default), so it should be gitignore'd.
+
+👉 **Add `.react-router/` to `.gitignore`**
+
+```txt
+.react-router/
+```
+
+You should also ensure that generated types are always present before running typechecking,
+especially for running typechecking in CI.
```

Comment:
Small tweak, I think just state the instruction instead of "you should" in general sounds good for this sort of a doc

```suggestion
Make sure generated types are always present before running type checking,
especially when running type checking in CI.
```